pandas/indexes/multi.py

```diff
@@ -2069,20 +2069,14 @@ def convert_indexer(start, stop, step, indexer=indexer, labels=labels):
         else:
 
             loc = level_index.get_loc(key)
-            if level > 0 or self.lexsort_depth == 0:
+            if isinstance(loc, slice):
+                return loc
+            elif level > 0 or self.lexsort_depth == 0:
                 return np.array(labels == loc, dtype=bool)
-            else:
-                # sorted, so can return slice object -> view
-                try:
-                    loc = labels.dtype.type(loc)
```

Comment:
And it is indeed a huge difference, that was one of the reasons of performance degradation in MultiIndex indexing: https://github.com/pandas-dev/pandas/pull/14551

But as you found the numpy issue, it is not really a bug (it would have different behaviour in certain cases eg with floats), you just have to be careful when using it.

You could also say it is `get_loc` that returns the wrong type :-)